project.md

```diff
@@ -7,62 +7,195 @@ good way to meet their needs.
-user? What kinds of errors might occur, and how might they be communicated to
-the user?_
+The main way I hope this project runs is that after the user creates a file with
+objects and keyframe controls, they can render the file which will result in
+HTML, CSS, and JavaScript files being created.
+The user can then open the main HTML file in any browser and it will be a
+working presentation!
+I believe the main error that will occur is invalid syntax.
+If the syntax is part of what I define for the structure of the presentation, I
+will ideally display an appropriate error message when the user tries to compile
+the program.
+If the syntax is something borrowed from CSS/JS, then I may let that compile and
+defer to the CSS/JS debugging tools to display those errors, but I'm not
+entirely sure how I will convert the user-defined styles and transitions yet.
+
+If I want to get ambitious, another output I would like to add is annotated
+versions of each component the user creates (also HTML/CSS/JS).
+The primary annotation could be an overlayed coordinate grid so the user can
+easily determine where visual features of their object lie.
+This would help when the user wants to align a specific visual point to another
+rather than aligning only based on the bounding box edges.
+The annotation could also show bounding boxes and alignment constraints which
+could help with debugging.
 
 ### Expressiveness
 
-_What should be easy to do in this language? What should be possible, but
-difficult? What should be impossible or very difficult?_
+It should be easy to create fluent web-based presentations.
+In contrast to other presentation softwares, it should be easy to keep select
+components visible when going between keyframes, it should be effortless to
+transition many different object styles rather than just object visibility, and
+it should be straightforward to incorporate any web media (animations,
+simulations, interactive components, etc.) into the presentation.
+Although possible, it should be more difficult to create a "slide deck" in this
+language than it would be in other software since doing that would kind of
+defeat the purpose of using this language.
+It should be impossible to use this program for things outside the domain like
+complex computations, creating software, or generally getting non-web-based
+outputs.
 
 ### Related work
 
-_Are there any other DSLs in this domain? If not, describe how you know there
-aren't and conjecture why not. If so, describe them and provide links. How well
-do they address the need? Are there any particularly admirable qualities of the
-language? Are there parts of the language you think could be improved?_
+From what I can find, nothing comes close to what I want.
+When I look for presentation softwares, every. single. one. uses a slide deck
+format and they only seem to differ in more minor features.
+Some examples of the existing softwares are PowerPoint, Google Slides, Prezi,
+Keynote, Beamer, etc.
```

Comment:
I might classify Beamer as an internal DSL for presentations, implemented in LaTeX.